server/src/main/java/org/apache/iotdb/db/query/dataset/NewEngineDataSetWithoutValueFilter.java

```diff
@@ -327,6 +328,167 @@ public TSQueryDataSet fillBuffer(int fetchSize, WatermarkEncoder encoder) throws
 
     return tsQueryDataSet;
   }
+  
+  /**
+   * for RPC in RawData query between client and server
+   * fill time buffer, value buffers and bitmap buffers
+   */
+  public TSQueryNonAlignDataSet fillNonAlignBuffer(int fetchSize, WatermarkEncoder encoder) throws IOException, InterruptedException {
```

Comment:
The fillNonAlignBuffer logic should be put into each single read task. The main thread just takes from the blocking queue and set it into QueryDataSet.

Reply:
May I ask what object will be in the blocking queue in this scenario? In each single read task, we get batch data from reader and transfer to bytebuffer. And then?

Reply:
Fixed